narwhals/_polars/series.py

```diff
@@ -407,3 +411,27 @@ def func(*args: Any, **kwargs: Any) -> Any:
             )
 
         return func
+
+
+class PolarsSeriesListNamespace:
+    def __init__(self: Self, series: PolarsSeries) -> None:
+        self._series = series
+
+    def len(self: Self) -> PolarsSeries:
+        native_series = self._series._native_series
+        if self._series._backend_version < (1, 16):  # pragma: no cover
+            native_result = native_series.list.len().zip_with(
+                ~native_series.is_null(), native_series.list.first()
+            )
```

Comment:
Probably there is a better way that using `native_series.list.first()`?

Reply:
how about
```python
pl.select(pl.when(s.is_not_null()).then(s.list.len()).otherwise(None))[s.name]
```
?

Reply:
TIL about `pl.select` 👌